src/server/ofonomodemmanager.cpp

```diff
@@ -113,7 +107,40 @@ bool OfonoModemManager::addModem(QString objectPath)
     return false;
 }
 
-QStringList OfonoModemManager::getModems()
+void OfonoModemManager::serviceRegistered()
+{
+    getModems();
+}
+
+void OfonoModemManager::getModems()
+{
+    QDBusMessage request = QDBusMessage::createMethodCall(OfonoConstants::OFONO_SERVICE,
+                                                          OfonoConstants::OFONO_MANAGER_PATH,
+                                                          OfonoConstants::OFONO_MANAGER_INTERFACE,
+                                                          "GetModems");
+    QDBusPendingReply<OfonoModemList> reply = QDBusConnection::systemBus().asyncCall(request);
+    QDBusPendingCallWatcher *watcher = new QDBusPendingCallWatcher(reply, this);
+    QObject::connect(watcher, SIGNAL(finished(QDBusPendingCallWatcher*)),
+                     this, SLOT(getModemsReply(QDBusPendingCallWatcher*)));
+}
+
+void OfonoModemManager::getModemsReply(QDBusPendingCallWatcher *call)
+{
+    QDBusReply<OfonoModemList> reply = *call;
+    call->deleteLater();
+    if (reply.error().isValid()) {
+        log_error("DBus call to interface %s function GetModems of path %s failed: %s",
+                  OfonoConstants::OFONO_MANAGER_INTERFACE,
+                  OfonoConstants::OFONO_MANAGER_PATH,
+                  reply.error().message().toStdString().c_str());
+    } else {
+        OfonoModemList list = reply;
+        for (int i = 0; i < list.count(); i++)
+            onModemAdded(list.at(i).name, QVariantMap());
```

Comment:
Yes, here we obtain the object path for modems, then we register listeners & query stuff from those modems in for example NetworkOperator and NetworkTime.